contrib/initramfs/scripts/zfs.in

```diff
@@ -424,6 +424,34 @@ decrypt_fs()
 			if ! [ "${KEYLOCATION}" = "prompt" ]; then
 				$ZFS load-key "${ENCRYPTIONROOT}"
 
+                        # Wait for input from named pipe, if Dropbear is running
+                        elif cat /run/dropbear.pid &>/dev/null; then
```

Comment:
Is it safe to assume if dropbear is ever running that it must be for ZFS root unlock?  If not, would setting a user property on `$ENCRYPTIONROOT` to signal intention be helpful?

Reply:
Good question.
We do can detect the presence of the attached GPU, because during the boot first active GPU assigned as active tty. For example on some hosters with KVM virtualization machines have these fake boot tty attached, they use this feature as a part of their recovery measures,  but the output goes into the void, unless the owner activated the boot console. Quemu and libvirt follows these rules.
Here is a basic working example on detecting attached GPU:
````
if find /sys/devices/ -name *boot*vga* | wc -l | grep  1 > /dev/null; then
  echo boot tty attached; 
fi
````
I've deliberately excluded this part from the zfs code, because I assume the user installing dropbear-initramfs package makes a conscious choice over accessing the system via SSH, The password request is activated only if zfs loader detects the encrypted partition. In non-standard situation the user has the ability to break the password request, drop into the SSH prompt, do necessary repair/rescue actions, then enter `zfsunlock` command to perform root pool decryption later. In any case, max 3 decryption attempts are accepted, then system forcibly reboots to delay brute password guessing.